README.md

```diff
@@ -242,7 +247,135 @@ TO DO:
 - [x] Verifies that the leaf used inside the circuit is equal to the `leaf` passed as (public) value to the instance column
 - [x] Add 2 public inputs to merkle_v1
 
+# Experiment 9 - Add carry v1
 
+Allowing the addition of new values to previously accumulated amounts into two columns, acc_hi and acc_lo.
 
+Circuit looks like this,
 
+| - | value  | acc_hi(x * 2^16)  | acc_lo(x * 2^0) | instance  |
+| - | ----      | ---      |   ---      | --|
+| 0 | - | 0 |  0xffff | 0 |
+| 1 | 0x1 | 0x1 |  0 | 0xffff |
+| 2 | - | - | - | 0x1 |
+| 3 | - | - | - | 0 |
 
+### Configuration 
+
+the first rows's values assigned with instance, that would be replace 0 with using assign from constant (or zero value from instance) in future overflow gadget. And `assign_row` function needs values from first row as arguments, these will be copied for the region. and then permutation check like below. 
+
+```Rust
+// following above table
+0 == (value + (acc_hi[1] * (1 << 16)) + acc_lo[1]) 
+    - ((acc_hi[2] * (1 << 16)) + acc_lo[2] )
+
+```
+
+`cargo test --package halo2-experiments --lib -- circuits::add_carry_v1`
+
+TO DO: -> moved to next version.
+
+~~- [ ] Range check for left most column of multi-columns for accumulation~~<br>
+~~- [ ] Support 2^256 in Accumulated value with multi-columns~~
+
+# Experiment 10 - Add carry v2
```

Comment:
The role of this modified chip, if I understood it correctly, is to check that the accumulator `acc_lo` doesn't overflow 16 bits. I think that the isZero constraint is not sufficient to prevent that, as nothing blocks a malicious prover from putting a value greater than 16 bits in `acc_lo` as there are no constraints in place to prevent that. I suggest removing the `isZero constraint` and replacing it with a `rangeCheck(16 BITS)` over the acc_lo column.

Reply:
True, It is. This chip does not have that constraint. So, I did not use this for `safe_accumulator`.
I simply want to show the possibility to check overflow by using the `isZero` constraint.
I should add a comment about it.